libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -2288,6 +2287,27 @@ void ExpressionCompiler::appendCompareOperatorCode(Token _operator, Type const&
 				m_context << Instruction::SWAP1;
 				m_context << ((u256(1) << 32) - 1) << Instruction::AND;
 			}
+			else if (funType->kind() == FunctionType::Kind::External)
+			{
+				// TO DO::
+				// The stack will look like <fun1_addr> <fun1_selector> <fun2_addr> <fun2_selector>
+				// In order to compare you need to swap fun2_selector with fun1_addr to move fun1_addr up.
+				// You can use swap3 for that.
+				m_context << Instruction::SWAP3;
+				// now the stack looks like <fun2_selector> <fun1_selector> <fun2_address> <fun1_address>
+				m_context << Instruction::EQ;
+				// now the stack looks like <fun2_selector> <fun1_selector> <TRUE/FALSE>;
+				m_context << Instruction::SWAP2;
+				// now the stack looks like <TRUE/FALSE> <fun1_selector> <fun2_selector>;
+				m_context << Instruction::EQ;
+				// now the stack looks like <TRUE/FALSE> <TRUE/FALSE>
+				m_context << Instruction::AND;
+				// now the stack looks lke <TRUE/FALSE>
+
+				// TO DO :: Figure out what the left shift operations are for
+				// Check and see if we need these operations in our case as well.
```

Comment:
Even though it takes two stack slots, the entire data can be fit into a single slot. It's a selector `4` bytes and address `20` bytes--24 bytes in total. A stack slot is 32 bytes. That's why, often, you see some shifts related to function types.

Reply:
That makes sense. But will we need to perform this shift here ? 
I have a feeling that if the (32 + 8 ) bytes of unoccupied slot space is garbage values then we'll need the shift. If , however, they've been initialized to 0 ( or null etc) , then we could do without the shift operator as well.